### PR TITLE
[Feeds] add CommunityMQTT feed

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot/constants.py [hunk 1]
```diff
@@ -44,12 +45,21 @@
 DEFAULT_COMMUNITY_WS_URL = "ws://community2.octobot.cloud/"   #TODO
 OCTOBOT_COMMUNITY_URL = os.getenv("COMMUNITY_SERVER_URL", DEFAULT_COMMUNITY_URL)
 OCTOBOT_COMMUNITY_FEED_URL = os.getenv("OCTOBOT_COMMUNITY_WS_URL", f"{DEFAULT_COMMUNITY_WS_URL}cable")
-OCTOBOT_COMMUNITY_AUTH_URL = f"{OCTOBOT_COMMUNITY_URL}spree_oauth/token"
+OCTOBOT_COMMUNITY_FEED_URL = os.getenv("OCTOBOT_COMMUNITY_MQTT_URL", "iot.fr-par.scw.cloud")    #TODO
```
Comment (GuillaumeDSM) [inline octobot/constants.py:48]: do we confirm this mqtt provider for beta version?

Comment (Herklos) [reply octobot/constants.py:48]: Yes

Comment (GuillaumeDSM) [reply octobot/constants.py]: Ok, I cleaned up all those constants

File: octobot/community/authentication.py [hunk 2]
```diff
@@ -412,4 +411,4 @@ def _reset_tokens(self):
 
     @staticmethod
     def _get_encoded_community_token():
-        return base64.encodebytes(constants.COMMUNITY_TOKEN.encode()).decode().strip()
+        return base64.encodebytes(constants.COMMUNITY_BACKEND_PUBLIC_TOKEN.encode()).decode().strip()
```
Comment (Herklos) [inline octobot/community/authentication.py:414]: 👍

File: octobot/community/feeds/community_ws_feed.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ async def _fetch_stream_identifier(self, identifier):
         if identifier is None:
             return None
         async with self.authenticator.get_aiohttp_session().get(
-                f"{constants.OCTOBOT_COMMUNITY_FETCH_FEED_IDENTIFIER_URL}/{identifier}") as resp:
+                f"{constants.OCTOBOT_COMMUNITY_URL}api/v2/storefront/feeds/id/{identifier}") as resp:
```
Comment (Herklos) [inline octobot/community/feeds/community_ws_feed.py:174]: That's a temporary url right?

Comment (GuillaumeDSM) [reply octobot/community/feeds/community_ws_feed.py:174]: It's a legacy one, we don't use `community_ws_feed` anymore

Comment (Herklos) [reply octobot/community/feeds/community_ws_feed.py:174]: Right

File: octobot/community/feeds/community_mqtt_feed.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        self.logger.debug(f"Sending message on topic: {topic}, message: {message}")
+        self._mqtt_client.publish(
+            self._build_topic(channel_type, identifier),
+            self._build_message(channel_type, message),
+            qos=self.default_QOS
+        )
+
+    def _get_callbacks(self, topic):
+        for callback in self.feed_callbacks.get(topic, ()):
+            yield callback
```
Comment (Herklos) [inline octobot/community/feeds/community_mqtt_feed.py:210]: 👍

File: octobot/community/feeds/community_mqtt_feed.py [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+        # from https://github.com/wialon/gmqtt/blob/master/examples/resubscription.py#L28
+        # in order to check if all the subscriptions were successful, we should first get all subscriptions with this
+        # particular mid (from one subscription request)
+        subscriptions = client.get_subscriptions_by_mid(mid)
+        for subscription, granted_qos in zip(subscriptions, qos):
+            # in case of bad suback code, we can resend  subscription
+            if granted_qos >= gmqtt.constants.SubAckReasonCode.UNSPECIFIED_ERROR.value:
+                self.logger.warning(f"Retrying subscribe, client_id: {client._client_id}, mid: {mid}, "
+                                    f"reason code: {granted_qos}, properties {properties}")
+                if self._subscription_attempts < self.MAX_SUBSCRIPTION_ATTEMPTS * len(subscriptions):
```
Comment (Herklos) [inline octobot/community/feeds/community_mqtt_feed.py:274]: 👍

File: octobot/community/graphql_requests.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+#  OctoBot is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public
+#  License along with OctoBot. If not, see <https://www.gnu.org/licenses/>.
+
+
+def create_new_device_query(user_id) -> (str, dict):
```
Comment (Herklos) [inline octobot/community/graphql_requests.py:18]: 👍

**Review by Herklos — APPROVED** [review]

**Comment by GuillaumeDSM** [comment]
to be merged after the next octobot_trading release (will be released after bitget tests)

**Review by Herklos — APPROVED** [review]

**Pull request merged**